salt/engines/stalekey.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/python
+
+'''
+An engine that uses presence detection to keep track of which minions
+have been recently connected and remove their keys if they have not been
+connected for a certain period of time.
+
+Requires that the minion_data_cache option be enabled.
+
+.. versionadded: Nitrogen
+
+:configuration:
+
+    Example configuration
+        engines:
+          - stalekey:
+              interval: 3600
+              expire: 86400
+
+'''
+
+import salt.utils.minions
+import salt.config
+import salt.key
+import salt.wheel
+import msgpack
+import os
+import time
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    opts = salt.config.master_config('/etc/salt/master')
```

Comment:
Ah, uh, no, I've just written so many utility scripts lately that don't run in Salt's context that I've gotten used to doing that. :)